test/cpp/interop/backend_metrics_lb_policy.cc

```diff
@@ -232,9 +260,34 @@ LoadReportTracker::GetNextLoadReport() {
   return report;
 }
 
+LoadReportTracker::LoadReportEntry LoadReportTracker::WaitForOobLoadReport(
+    const std::function<bool(const xds::data::orca::v3::OrcaLoadReport&)>&
+        predicate,
+    absl::Duration poll_timeout, size_t max_attempts) {
+  absl::MutexLock lock(&load_reports_mu_);
+  // This condition will be called under lock
+  auto condition = [&]() ABSL_NO_THREAD_SAFETY_ANALYSIS {
+    return !oob_load_reports_.empty();
+  };
+  for (size_t i = 0; i < max_attempts; i++) {
+    if (!load_reports_mu_.AwaitWithTimeout(absl::Condition(&condition),
+                                           poll_timeout)) {
+      return absl::nullopt;
+    }
+    auto report = oob_load_reports_.front();
+    oob_load_reports_.pop_front();
+    if (predicate(report)) {
+      gpr_log(GPR_DEBUG, "Report #%zu matched", i + 1);
```

Comment:
I think `%z` is not a portable format specifier.  You need to use `PRIuPTR` instead.